src/sinks/aws_cloudwatch_metrics.rs

```diff
@@ -177,38 +184,80 @@ impl RetryLogic for CloudWatchMetricsRetryLogic {
         match error {
             PutMetricDataError::HttpDispatch(_) => true,
             PutMetricDataError::InternalServiceFault(_) => true,
-            PutMetricDataError::Unknown(res) if res.status.is_server_error() => true,
+            PutMetricDataError::Unknown(res)
+                if res.status.is_server_error()
+                    || res.status == http::StatusCode::TOO_MANY_REQUESTS =>
+            {
+                true
+            }
             _ => false,
         }
     }
 }
 
-fn encode_events(events: Vec<Event>, namespace: String) -> Result<PutMetricDataInput, ()> {
+fn timestamp_to_string(timestamp: DateTime<Utc>) -> String {
```

Comment:
We could make this fn public https://github.com/timberio/vector/blob/master/src/event/mod.rs#L281 and use that one instead of redefining it?

Reply:
I'm not sure, because there is a difference in SecondsFormat: AutoSi vs Millis.

The CloudWatch spec for timestamp is vague. It contains a knows issue of documenting this field such as "expressed as the number of milliseconds since Jan 1, 1970 00:00:00 UTC", however it's not really true. The examples and tooling around this API all seems to use ms precision, and I've decided to do the same to feel safer.

Reply:
Ok that is fine, I probably didn't notice that detail 😄